scripts/gitlab/check_runtime.sh

```diff
@@ -52,176 +52,178 @@ runtime_regex="^runtime/$(join_by '|^runtime/' "${runtimes[@]}" "${common_dirs[@
+      github_label "D2-breaksapi"
+    fi
 
-			changes to the ${RUNTIME} runtime sources and changes in the spec version.
+    boldcat <<EOT
+## RUNTIME: ${RUNTIME} ##
 
-			spec_version: ${sub_spec_version} -> ${add_spec_version}
+changes to the ${RUNTIME} runtime sources and changes in the spec version.
 
-		EOT
-		continue
+spec_version: ${sub_spec_version} -> ${add_spec_version}
 
-	else
-		# check for impl_version updates: if only the impl versions changed, we assume
-		# there is no consensus-critical logic that has changed.
+EOT
+    continue
 
-		add_impl_version="$(
-			git diff refs/tags/"${LATEST_TAG}...${CI_COMMIT_SHA}" "runtime/${RUNTIME}/src/lib.rs" \
-			| sed -n -r 's/^\+[[:space:]]+impl_version: +([0-9]+),$/\1/p'
-		)"
-		sub_impl_version="$(
-			git diff refs/tags/"${LATEST_TAG}...${CI_COMMIT_SHA}" "runtime/${RUNTIME}/src/lib.rs" \
-			| sed -n -r 's/^\-[[:space:]]+impl_version: +([0-9]+),$/\1/p'
-		)"
+  else
+    # check for impl_version updates: if only the impl versions changed, we assume
+    # there is no consensus-critical logic that has changed.
 
+    add_impl_version="$(
+      git diff refs/tags/"${LATEST_TAG}...${CI_COMMIT_SHA}" "runtime/${RUNTIME}/src/lib.rs" \
+      | sed -n -r 's/^\+[[:space:]]+impl_version: +([0-9]+),$/\1/p'
+    )"
+    sub_impl_version="$(
+      git diff refs/tags/"${LATEST_TAG}...${CI_COMMIT_SHA}" "runtime/${RUNTIME}/src/lib.rs" \
+      | sed -n -r 's/^\-[[:space:]]+impl_version: +([0-9]+),$/\1/p'
+    )"
 
-		# see if the impl version changed
-		if [ "${add_impl_version}" != "${sub_impl_version}" ]
-		then
-			boldcat <<-EOT
 
-			## RUNTIME: ${RUNTIME} ##
+    # see if the impl version changed
+    if [ "${add_impl_version}" != "${sub_impl_version}" ]
+    then
+      boldcat <<EOT
 
-			changes to the ${RUNTIME} runtime sources and changes in the impl version.
+## RUNTIME: ${RUNTIME} ##
 
-			impl_version: ${sub_impl_version} -> ${add_impl_version}
+changes to the ${RUNTIME} runtime sources and changes in the impl version.
 
-			EOT
-			continue
-		fi
+impl_version: ${sub_impl_version} -> ${add_impl_version}
 
+EOT
+      continue
+    fi
 
-		boldcat <<-EOT
-		wasm source files changed or the spec version in the substrate reference in
-		the Cargo.lock but not the spec/impl version. If changes made do not alter
-		logic, just bump 'impl_version'. If they do change logic, bump
-		'spec_version'.
+    failed_runtime_checks+=($RUNTIME)
+  fi
+done
 
-		source file directories:
-		- runtime
+if [ ${#failed_runtime_checks} -gt 0 ]; then
+  boldcat <<EOT
+wasm source files changed or the spec version in the substrate reference in
+the Cargo.lock but not the spec/impl version. If changes made do not alter
+logic, just bump 'impl_version'. If they do change logic, bump
+'spec_version'.
 
-		versions file: ${VERSIONS_FILE}
+source file directories:
+- runtime
 
-		EOT
+version files: ${failed_runtime_checks[@]}
+EOT
 
-		exit 1
-	fi
-done
+  exit 1
+fi
```

Comment:
I added this change to this PR. It's not really related, I just wanted to fix it.

Instead of failing at the first runtime check, it now collects all the runtime check that fails and print them in a single build.

I also converted the file from tabs to spaces because all other bash scripts in this repo are using spaces.